hooks/analytics/useTrackingEvents.js

```diff
@@ -0,0 +1,22 @@
+import { useRouter } from "next/router";
+import { useAnalytics } from "use-analytics";
+import trackProduct from "lib/tracking/trackProduct";
+
+
+/**
+ * tracking events visits
+ *
+ * @param {String} data - the props of the page
+ * @returns {Object} useTrackingEvents
+ */
+export default function useTrackingEvents() {
```

Comment:
Regarding the bigger picture, I'm a bit unclear on that one.
This is called `useTrackingEvents`, but it seems to cover no shared logic relevant for tracking the various events other than the `trackProduct` event?

Also, if we end up adding more of them here and maybe rename `callTracking`, the hook would get a bit bloated and declare several functions during render for each components somewhere making use one of the tracking events.

Maybe instead, call this one something like `useTrackProducts` which returns some `trackProduct` function?
To then give that hook some more purpose, import the constant `PRODUCT_VIEWED` right here instead of passing it as part of the payload from the `ProductDetail` component into the function.

Optionally, also include the functionality imported from `lib/tracking/trackProduct` right here as AFAIK we have no other usecase for that, but that's just an idea.

I don't want to be too picky about this, but I'd really want the result of this to be less cubersome and easier to understand than the previous implementation.

Reply:
is that a good idea or instead of doing that I can create a custom hook to each event (exactly as you said) and transform ProductDetail to a functional component?

Reply:
My general preference on this is, that we should keep as much of the non ui-specific logic out of the component as possible, and put that behind some hook - If you don‘t want to convert a class components to functional to be able to use the hook, I‘m fine with some HOC wrapper using the hook like you did here for the pdp component already.

Regarding the `useTrackingEvents` hook, I think it‘s fine to keep it if it provides some general logic that can be used across multiple types of tracking events, but then I would make the specific event hook, e.g. useProductTrackingEvent, make use of that hook.

Does that make sense to you the way I put it? :)

Reply:
Hi, yes cristal clear :D 

I decided to refactor the ProductDetail component to a Functional component.